src/Illuminate/Database/Query/Builder.php

```diff
@@ -1420,16 +1420,28 @@ public function simplePaginate($perPage = 15, $columns = ['*'])
 	/**
 	 * Get the count of the total records for the paginator.
 	 *
+	 * @param  array  $columns
 	 * @return int
 	 */
-	public function getCountForPagination()
+	public function getCountForPagination($columns = ['*'])
 	{
 		$this->backupFieldsForCount();
 
-		$total = $this->count();
+		$this->aggregate = ['function' => 'count', 'columns' => $columns];
+
+		$results = $this->get();
+
+		$this->aggregate = null;
 
 		$this->restoreFieldsForCount();
 
+		if (isset($results[0]))
+		{
+			if (isset($this->groups)) return count($results);
+
+			return (int) array_change_key_case((array) $results[0])['aggregate'];
+		}
+
 		return $total;
```

Comment:
This line should be removed. Moreover, the method may now return null. I would change it all as follows:

```php
// ...
$this->restoreFieldsForCount();

if (isset($this->groups)) return count($results);

return isset($results[0]) ? (int) array_change_key_case((array) $results[0])['aggregate'] : 0;
```